scripts/nesi_scripts/adv_im/run_adv_im.sh

```diff
@@ -0,0 +1,209 @@
+echo "linking Data folder"
+ln -s $SIM_DATA_DIR $root_dir
+
+################################################################
+#   determine mag catagory
+################################################################
+max_mag=`grep mag $root_dir/Data/VMs/*/vm_params.yaml | awk '{print $2}' | sort -unr | head -n 1`
+if (( $(echo "$max_mag > 7.0" | bc -l) ));then
+    mag_category=large
+elif (( $(echo "$max_mag > 5.0" | bc -l) ));then
+    mag_category=moderate
+else
+    mag_category=small
+fi
+
+# install the simulation
+echo $SPLIT_LINE
+echo "installing simulation folder"
+echo $SPLIT_LINE
+python $gmsim/workflow/scripts/cybershake/install_cybershake.py $root_dir $LIST_EVENTS_F $gmsim_version --stat_file_path $STATION_LIST
+if [[ $? != 0 ]];then
+    exit
+fi
+echo $SPLIT_LINE
+
+################################################################
+# add/update the adv_im models in root_params.yaml
+python $gmsim/workflow/scripts/update_root_yaml.py $root_dir/Runs/root_params.yaml $ADV_IM_NAME
+
+if [[ $? != 0 ]];then
+    exit 5
+fi
+################################################################
+
+# link BB
+echo $SPLIT_LINE
+echo "linking BB"
+bash $gmsim/workflow/scripts/link_bb.sh $BBbin_root_dir $root_dir $LIST_EVENTS_F
+
+if [[ $? != 0 ]];then
+    exit 
+fi
+
+# link observed data
+echo $SPLIT_LINE
+echo "linking ObservedData"
+bash $gmsim/workflow/scripts/link_obs.sh $OBS_DATA_DIR $root_dir $LIST_EVENTS_F
+
+if [[ $? != 0 ]];then
+    exit 
+fi
+
+# cp task_config.yaml
+if [[ -f $TASK_CONFIG ]];then
+    rm $TASK_CONFIG
+fi
+cp $TASK_CONFIG_TEMPLATE $TASK_CONFIG
+
+# run a test run on HALS station to get the time ratio for estimating WCT
+# comparing to ATC12_story
+echo $SPLIT_LINE
+echo "Running test station to estimate WCT"
+echo "using waveform from $mag_category event"
+tmp_test_dir=$root_dir/test_tmp
+if [[ ! -d $tmp_test_dir ]]; then
+    mkdir -p $tmp_test_dir
+fi
+TIMEFORMAT='%R'
+used_time="$(time (bash $gmsim/test_station/test_runpy.sh $ADV_IM_NAME $mag_category $tmp_test_dir) 2>&1 1>/dev/null)"
+if [[ $? != 0 ]];then
+    exit 10
+fi
+echo "used time: $used_time"
+rm -r $tmp_test_dir
+
+est_ratio=` echo $used_time / $WCT_BENCHMARK_MINSEC | bc -l`
+echo "est_ratio = $est_ratio"
+if (( $( echo "$est_ratio > 1" | bc -l ) ));then
+    # up-scale the WCT
+    sed -i "/threshold_time = /c\    threshold_time = $WCT_BENCHMARK_MAXHOUR*$est_ratio" $gmsim/workflow/estimation/estimate_wct.py
+else
+    sed -i "/threshold_time = /c\    threshold_time = $WCT_BENCHMARK_MAXHOUR" $gmsim/workflow/estimation/estimate_wct.py
+fi
+
+
+# create screen socket and run automated workflow
+function create_runsocket {
+    screen -d -m -S run_$ADV_IM_NAME bash -c "python $gmsim/workflow/scripts/cybershake/run_cybershake.py $root_dir $USERNAME $TASK_CONFIG --n_max_retries 1"
+}
+create_runsocket
+while [[ $? != 0 ]]; 
+do
+    echo "failed to start up screen socket, waiting 5 sec to attemp again"
+    sleep 5
+    create_runsocket
+done
```

Comment:
Max retry counter? Otherwise this could go forever

Reply:
attempt

Reply:
my original thought was that the user should decide when they are going to kill it or not.
this error (failed to start a screen) rarely happens, and my guess was that its related to if anyone is stressing the login node, thus usually naturally resolve itself after a few try. ( hence the unconditioned loop until success)